pkg/target/host.go

```diff
@@ -214,14 +209,18 @@ func (h *Host) Connect(attemptCount int) {
 			zap.Uint16("port", h.Port))
 		h.Conn = nil
 		if attemptCount == 1 {
-			h.updateHostHealthStatus <- &HostStatus{Host: h, Status: false, sigCh: make(chan struct{})}
+			if h.Available.Load() {
```

Comment:
```suggestion
			if h.Available.CAS(true, false) {
```

and then we dont' need `h.Available.Store(false)`